### ISSUE TITLE
Reboot maps volume to /var/lib

### ISSUE THREAD

**Issue by tobad357** [hpe-storage/csi-driver#65]
When a node comes back up from a reboot or a hard reset our volumes get mapped to /var/lib volume instead of to the proper multipath volume

```
cat /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-7b735c41-37c7-4505-9140-22ca514c546a/globalmount/deviceInfo.json 
{"volume_id":"0671136b496958f3f500000000000000000000009a","volume_access_mode":2,"device":{"path_name":"dm-11","serial_number":"2069cb61307eb56ea6c9ce900bc984b18","major":"253","minor":"11","alt_full_path_name":"/dev/mapper/mpathm","mpath_device_name":"mpathm","size":102400,"slaves":["sdh"],"iscsi_target":{"Name":"iqn.2007-11.com.nimblestorage:pvc-7b735c41-37c7-4505-9140-22ca514c546a-v71136b496958f3f5.0000009a.184b98bc","Address":"10.32.8.7","Port":"3260","Tag":"2460","Scope":""},"target_scope":"volume","state":"active"},"mount_info":{"mount_point":"/var/lib/kubelet/plugins/hpe.com/mounts/0671136b496958f3f500000000000000000000009a","filesystem_options":{"Type":"xfs","Mode":"","Owner":""}}}
```

```
mount |grep 7b73
/dev/mapper/centos00-var_lib on /var/lib/kubelet/pods/1787c1da-969e-4f09-b158-ab683014c788/volumes/kubernetes.io~csi/pvc-7b735c41-37c7-4505-9140-22ca514c546a/mount type xfs (rw,relatime,seclabel,attr2,inode64,logbsize=256k,sunit=512,swidth=512,noquota)
```

```
ls -l /dev/mapper/
total 0
lrwxrwxrwx. 1 root root       7 Nov  3 04:49 centos00-home -> ../dm-5
lrwxrwxrwx. 1 root root       7 Nov  3 04:49 centos00-root -> ../dm-0
lrwxrwxrwx. 1 root root       7 Nov  3 04:49 centos00-swap -> ../dm-1
lrwxrwxrwx. 1 root root       7 Nov  3 04:49 centos00-var -> ../dm-2
lrwxrwxrwx. 1 root root       7 Nov  3 04:49 centos00-var_lib -> ../dm-4
lrwxrwxrwx. 1 root root       7 Nov  3 04:49 centos00-var_log -> ../dm-3
crw-------. 1 root root 10, 236 Nov  3 04:49 control
lrwxrwxrwx. 1 root root       7 Nov  3 04:51 mpathh -> ../dm-7
lrwxrwxrwx. 1 root root       7 Nov  3 04:51 mpathi -> ../dm-8
lrwxrwxrwx. 1 root root       7 Nov  3 04:51 mpathj -> ../dm-6
```

Looking at the iscsiadm session shows there is no session for this volume and it seems like the driver then uses /var/lib device

csi driver version
```
v1.0.0-beta: Pulling from hpestorage/csi-driver
Digest: sha256:89e9d0fd81b09eba0db567114962bd9eb957cac62f134e07712240605b75e9a6
Status: Image is up to date for hpestorage/csi-driver:v1.0.0-beta
docker.io/hpestorage/csi-driver:v1.0.0-beta
```

**Comment by tobad357** [comment]
node csi driver logs

```
time="2019-11-03T04:51:17Z" level=trace msg="Found Volume &{ID:0671136b496958f3f500000000000000000000009a Name:pvc-7b735c41-37c7-4505-9140-22ca514c546a Size:107374182400 Description:System Standard Storage on Nimble01 InUse:false Published:true BaseSnapID: ParentVolID: Clone:false Config:map[clone_of: create_snapshot:false dedupe_enabled:false destroy_on_delete:false encrypted:true folder:system limit_iops:-1 limit_mbps:-1 performance_policy:default pool:default sync_on_detach:false target_scope:volume thick:false] Metadata:[] SerialNumber: AccessProtocol: Iqn: DiscoveryIP: MountPoint: Status:map[] Chap:<nil> Networks:[] ConnectionMode: LunID: TargetScope: IscsiSessions:[] FcSessions:[]}" file="driver.go:363"
time="2019-11-03T04:51:17Z" level=trace msg="<<<<< GetVolume" file="driver.go:364"
time="2019-11-03T04:51:17Z" level=trace msg=">>>>> isVolumeStaged, volumeID: 0671136b496958f3f500000000000000000000009a, stagingTargetPath: /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-7b735c41-37c7-4505-9140-22ca514c546a/globalmount, stagingMountPoint: /var/lib/kubelet/plugins/hpe.com/mounts/0671136b496958f3f500000000000000000000009a, volAccessType: mount" file="node_server.go:271"
time="2019-11-03T04:51:17Z" level=trace msg=">>>>> FileExists for path /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-7b735c41-37c7-4505-9140-22ca514c546a/globalmount/deviceInfo.json" file="file.go:62"
time="2019-11-03T04:51:17Z" level=trace msg="<<<<< FileExists" file="file.go:72"
time="2019-11-03T04:51:17Z" level=trace msg=">>>>> readStagedDeviceInfo, targetPath: /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-7b735c41-37c7-4505-9140-22ca514c546a/globalmount" file="node_server.go:1615"
time="2019-11-03T04:51:17Z" level=trace msg=">>>>> FileExists for path /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-7b735c41-37c7-4505-9140-22ca514c546a/globalmount/deviceInfo.json" file="file.go:62"
time="2019-11-03T04:51:17Z" level=trace msg="<<<<< FileExists" file="file.go:72"
time="2019-11-03T04:51:17Z" level=trace msg="<<<<< readStagedDeviceInfo" file="node_server.go:1644"
time="2019-11-03T04:51:17Z" level=trace msg="Found staged device: &{VolumeID:0671136b496958f3f500000000000000000000009a VolumeAccessMode:mount POD:<nil> Secret:<nil> Device:0xc0005461c0 MountInfo:0xc000306e10}" file="node_server.go:287"
time="2019-11-03T04:51:17Z" level=trace msg=">>>>> GetMountsForDevice, device: &{ dm-11 2069cb61307eb56ea6c9ce900bc984b18 253 11 /dev/mapper/mpathm mpathm 102400 [sdh] 0xc0001e7180 [] volume active }" file="chapidriver_linux.go:227"
time="2019-11-03T04:51:17Z" level=trace msg=">>>>> GetMountPointsForDevices, devices: [0xc0005461c0]" file="mount.go:89"
time="2019-11-03T04:51:17Z" level=trace msg=">>>>> FileGetStringsWithPattern called with path: /proc/mounts Pattern: " file="file.go:112"
time="2019-11-03T04:51:17Z" level=trace msg="<<<<< FileGetStringsWithPattern" file="file.go:135"
time="2019-11-03T04:51:17Z" level=trace msg="Checking mount for device &{VolumeID: Pathname:dm-11 SerialNumber:2069cb61307eb56ea6c9ce900bc984b18 Major:253 Minor:11 AltFullPathName:/dev/mapper/mpathm MpathName:mpathm Size:102400 Slaves:[sdh] IscsiTarget:0xc0001e7180 Hcils:[] TargetScope:volume State:active Filesystem:} " file="mount.go:106"
time="2019-11-03T04:51:17Z" level=trace msg="Checking partition for device &{VolumeID: Pathname:dm-11 SerialNumber:2069cb61307eb56ea6c9ce900bc984b18 Major:253 Minor:11 AltFullPathName:/dev/mapper/mpathm MpathName:mpathm Size:102400 Slaves:[sdh] IscsiTarget:0xc0001e7180 Hcils:[] TargetScope:volume State:active Filesystem:}" file="mount.go:119"
time="2019-11-03T04:51:17Z" level=trace msg="<<<<< GetMountPointsForDevices" file="mount.go:148"
time="2019-11-03T04:51:17Z" level=trace msg="<<<<< GetMountsForDevice" file="chapidriver_linux.go:231"
time="2019-11-03T04:51:17Z" level=trace msg="Found 0 mounts for device with serial number 2069cb61307eb56ea6c9ce900bc984b18" file="node_server.go:302"
time="2019-11-03T04:51:17Z" level=trace msg=">>>>> getMountInfo, volumeID: 0671136b496958f3f500000000000000000000009a, mountPath: /var/lib/kubelet/plugins/hpe.com/mounts/0671136b496958f3f500000000000000000000009a" file="node_server.go:36"
time="2019-11-03T04:51:17Z" level=trace msg="<<<<< getMountInfo" file="node_server.go:56"
time="2019-11-03T04:51:17Z" level=trace msg="Checking for mount options compatibility: staged options: [], reqMountOptions: []" file="node_server.go:316"
time="2019-11-03T04:51:17Z" level=trace msg="<<<<< isVolumeStaged" file="node_server.go:331"
time="2019-11-03T04:51:17Z" level=info msg="Volume 0671136b496958f3f500000000000000000000009a has already been staged. Returning here" file="node_server.go:220"
time="2019-11-03T04:51:17Z" level=trace msg="<<<<< nodeStageVolume" file="node_server.go:221"
time="2019-11-03T04:51:17Z" level=trace msg=">>>>> ClearRequest, key: NodeStageVolume:0671136b496958f3f500000000000000000000009a:/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-7b735c41-37c7-4505-9140-22ca514c546a/globalmount" file="driver.go:499"
time="2019-11-03T04:51:17Z" level=trace msg="Acquiring mutex lock for NodeStageVolume:0671136b496958f3f500000000000000000000009a:/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-7b735c41-37c7-4505-9140-22ca514c546a/globalmount" file="concurrent.go:44"
time="2019-11-03T04:51:17Z" level=trace msg="Print RequestCache: map[]" file="driver.go:519"
time="2019-11-03T04:51:17Z" level=trace msg="Successfully removed an entry with key NodeStageVolume:0671136b496958f3f500000000000000000000009a:/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-7b735c41-37c7-4505-9140-22ca514c546a/globalmount from the cache map" file="driver.go:520"
time="2019-11-03T04:51:17Z" level=trace msg="Releasing mutex lock for NodeStageVolume:0671136b496958f3f500000000000000000000009a:/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-7b735c41-37c7-4505-9140-22ca514c546a/globalmount" file="concurrent.go:67"
time="2019-11-03T04:51:17Z" level=trace msg="<<<<< ClearRequest" file="driver.go:521"
time="2019-11-03T04:51:17Z" level=trace msg="<<<<< NodeStageVolume" file="node_server.go:146"
```

**Comment by tobad357** [comment]
One interesting thing is that the device ```Pathname:dm-11``` doesn't exist on the server and neither does ```/dev/mapper/mpathm```

**Comment by tobad357** [comment]
Close as fixed in #66
